.github/workflows/create_release_pr.yml

```
@@ -0,0 +1,57 @@
+name: Create Release PR
+
+on:
+  workflow_dispatch:
+    inputs:
+      versionName:
+        description: 'Version to release (e.g., 2.1.0)'
+        required: true
+        type: string
+
+jobs:
+  create_release_pr:
+    name: Create Release PR
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0  # Full git history to ensure branch creation works
+
+      - name: Set up Java
+        uses: actions/setup-java@v4
+        with:
+          java-version: '17'
+          distribution: 'microsoft'
+
+      - name: Set github user
+        uses: ./.github/actions/set_github_user
+
+      - name: Update Version Files
+        run: |
+          # Update version in build.gradle
+          ./gradlew -PversionParam=${{ github.event.inputs.versionName }} changeReleaseVersion
+          
+          # Update CHANGELOG with the version and date
+          DATE=$(date +%Y-%m-%d)
+          ./gradlew -PversionParam=${{ github.event.inputs.versionName }} updateCHANGELOGVersion
+
+      - name: Create Pull Request
+        uses: peter-evans/create-pull-request@v5
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          commit-message: "Prepare release ${{ github.event.inputs.versionName }}"
+          branch: release/${{ github.event.inputs.versionName }}
+          delete-branch: false
```

```suggestion
          delete-branch: true
```
Would it make sense to delete the branch afterward? We may end up with a bunch of dangling refs. We do subsequently tag the branch before release too, so we shouldn't need the branch after it's been merged.